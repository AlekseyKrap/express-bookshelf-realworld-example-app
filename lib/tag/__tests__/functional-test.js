'use strict';

const chance = require('chance').Chance('tag');
const request = require('supertest');
const Tag = require('../model');
const app = require('../../app');

describe('tag', () => {
  let tagOne;
  let tagTwo;

  beforeEach(async () => {
    tagOne = await new Tag({name: chance.word()}).save();
    tagTwo = await new Tag({name: chance.word()}).save();
  });

  afterEach(async () => {
    await tagOne.destroy();
    await tagTwo.destroy();
  });

  describe('GET /api/tags', () => {
    test('should return a 200', async () => {
      const response = await request(app)
        .get('/api/tags')
        .set('Accept', 'application/json')
        .send();

      expect(response.statusCode).toBe(200);
      expect(response.body).toMatchSnapshot();
    });
  });
});
