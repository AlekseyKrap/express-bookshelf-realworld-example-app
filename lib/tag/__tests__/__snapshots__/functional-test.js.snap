// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag GET /api/tags should return a 200 1`] = `
Object {
  "tags": Array [
    "ehupezug",
    "jazas",
  ],
}
`;
